shared/types/user.ts

```diff
@@ -57,6 +57,7 @@ export const UserDto = UserBase.extend({
   created: DateString,
   lastAccessed: DateString.optional(),
   updatedAt: DateString,
+  grantSource: z.enum(['sso', 'otp']).optional(),
```

Comment:
this [is only set](https://github.com/opengovsg/FormSG/pull/8576/files#diff-408ae50a9d1df9342f01c80002c6fdee00849358ac045c4f9f99e78afa0cc2caR135) if we're logging in with sso:
- sgid doesn't currently set it
- otp doesn't currently set it

this flag is just used to display the [toast onLogout](https://github.com/opengovsg/FormSG/pull/8576/files#diff-a589bd616355733a26d62767cb2131ad05e0258f3de1c06cf893910bf93b6f7bR233-R240):
for directing SSO users who log out of FormSG to also logout on SSO